### PR TITLE
Fixes the login loop

### PR DESCRIPTION
**Pull request by bdc34**
This fixes the login loop that often happens when a user has logged into arxiv.org and then needs to log into beta.arxiv.org. Requests with more than one cookie for tapir_session would trigger this bug. Request like this are common because we run web apps at the arxiv.org domain and the beta.arxiv.org subdomain. Browsers will send cookies from both arxiv.org and beta.arxiv.org to beta.arxiv.org.
 
The cause of the loop in arxiv-auth was a method called detect_and_clobber_dup_cookies() that would detect duplicate cookie. If duplicates were detected arxiv-auth would redirect to the /login page. Since no possible response from a web app a beta.arxiv.org could never clobber the browsers cookies for arxiv.org this never worked. 

The solution here is to get all the duplicate cookies and use the first one that validates for the auth.
 
This fix need to be deployed along side a similar fix in arxiv-submit.   

https://arxiv-org.atlassian.net/browse/ARXIVNG-2784

### CODE REVIEW

**Comment by bmaltzan** [comment]
> The solution here is to get all the duplicate cookies and use the first one that validates for the auth.
That sounds like we could have a cant-logout event. I didn't find one though.

**Review by bmaltzan — APPROVED** [review]
Looks good.

**Comment by bdc34** [comment]
> > The solution here is to get all the duplicate cookies and use the first one that validates for the auth.
> 
> That sounds like we could have a cant-logout event. I didn't find one though.

That is a good thing to consider. It makes sense that the /logout page needs to be served from exactly the same domain name as the /login page so that the logout has the ability to clear the cookies.

**Comment by DavidLFielding** [comment]
Looks good!

**Review by DavidLFielding — APPROVED** [review]
Approved!

I believe this bug impacted developers the most as we tend to jump around to various arXiv servers.

File: users/arxiv/users/auth/__init__.py [hunk 1]
```diff
@@ -131,18 +127,13 @@ def load_session(self) -> Optional[Response]:
...
-            logger.debug('No session; attempting to get legacy session')
-            response = self.detect_and_clobber_dupe_cookies()
-
-            # Return that is anything other than None is treated as a response;
-            # request handling stops here.
-            if response is not None:
-                return response
-            session = self._get_legacy_session()
+            logger.debug('No session; attempting to get legacy from cookies')
+            session = next(filter(bool,
```
Comment (mhl10) [inline users/arxiv/users/auth/__init__.py:133]: Neat! Some brief documentation would be helpful here for future readers to unpack this.

**Review by mhl10 — APPROVED** [review]

File: users/arxiv/users/auth/__init__.py (hunk 1 above)
Comment (bdc34) [reply users/arxiv/users/auth/__init__.py]: I moved that code out to a method.

**Comment by bdc34** [comment]
I've increased the docker startup timeout to see if that fixes Travis. Looking at Travis I didn't see any news about changes how they use docker.

**Comment by bdc34** [comment]
> Approved!
> 
> I believe this bug impacted developers the most as we tend to jump around to various arXiv servers.

The real problem was that it looked sloppy when we had to ask beta testers, who are often mods, to "clear all your cookies" so they could log in to beta.arxiv.org.

**Pull request merged**